# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/dom_localization/test_domloc.xhtml
# Commit: 595529cd906f
# Full Hash: 595529cd906f94cdb2bc12a7a40072db0c07198e
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-03 22:01:19
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 9f3e2963d925 -r 595529cd906f dom/l10n/tests/mochitest/dom_localization/test_domloc.xhtml
--- a/dom/l10n/tests/mochitest/dom_localization/test_domloc.xhtml	Tue Aug 03 05:52:03 2021 +0000
+++ b/dom/l10n/tests/mochitest/dom_localization/test_domloc.xhtml	Tue Aug 03 05:52:03 2021 +0000
@@ -10,9 +10,9 @@
   <script src="chrome://mochikit/content/tests/SimpleTest/SimpleTest.js" />
   <script type="application/javascript">
   <![CDATA[
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource(`
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
 file-menu =
     .label = File
     .accesskey = F
@@ -20,9 +20,10 @@
     .label = New Tab
     .accesskey = N
 container = Some text with an <image data-l10n-name="foo"> inside it.
-`));
-    yield bundle;
-  }
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   SimpleTest.waitForExplicitFinish();
 
@@ -30,11 +31,12 @@
   const domLoc = new DOMLocalization(
     [],
     false,
-    { generateBundles },
+    l10nReg,
+    ["en-US"],
   );
 
   async function foo() {
-    domLoc.addResourceIds(["dummy.ftl"]);
+    domLoc.addResourceIds(["/mock.ftl"]);
     domLoc.connectRoot(document.documentElement);
     await domLoc.translateRoots();
 